x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/hooks/form.tsx

```diff
@@ -386,9 +386,13 @@ export function useOnSubmit({
       if (!error) {
         setSavedPackagePolicy(data!.item);
 
+        const newAgentlessPolicy = agentPolicies.find(
+          (policy: AgentPolicy) => policy?.supports_agentless === true
+        );
+
         const promptForAgentEnrollment =
           !(agentCount && agentPolicies.length > 0) &&
-          !isAgentlessPackagePolicy(data!.item) &&
+          !isAgentlessAgentPolicy(newAgentlessPolicy) &&
```

Comment:
nice, that's going to handle both serverless and ESS

Reply:
Yes! Thanks for catching this!